packages/dashboard/src/components/tests/dispensers-panel.test.tsx

```diff
@@ -14,8 +17,16 @@ beforeEach(() => {
 });
 
 it('renders dispensers', () => {
-  const root = mount(<DispenserPanel dispenserStates={dispenserStates} />);
+  const resources = new ResourceManager(fakeResources());
+  const root = mount(
+    <ResourcesContext.Provider value={resources}>
+      <DispenserPanel dispenserStates={dispenserStates} />
+    </ResourcesContext.Provider>,
+  );
   const dispenserElements = root.find(DispenserItem);
-  expect(dispenserElements.length).toBe(Object.keys(dispenserStates).length);
+  const dispenersInResource: number = resources.dispensers
+    ? Object.keys(resources.dispensers.dispensers).length
+    : 0;
+  expect(dispenserElements.length).toBe(dispenersInResource);
```

Comment:
A related comment (but not for now, it's ok at it is now), in the very near future we are going to change Enzyme to RTL, as well as philosophy, for which we should be verifying by the `Unknown` text and not only by quantity.